src/CoreLib/DSeq.fs

```diff
@@ -1011,10 +1041,14 @@ and [<AllowNullLiteral>]
 //            | :? MemStream as ms -> 
                 let ms = streamObject
                 // Add serial and numElems as prefix. 
-                let msPrefix = new MemStream( 128 )
+                //let msPrefix = new MemStream( 128 )
+                let msPrefix = ms.GetNew()
                 msPrefix.WriteInt64( meta.Serial )
                 msPrefix.WriteVInt32( meta.NumElems ) 
-                let msCombine = ms.InsertBefore( msPrefix )
+                //let msCombine = ms.InsertBefore( msPrefix )
+                ms.InsertBefore(msPrefix) |> ignore
+                ms.DecRef()
```

Comment:
Can we wrap the GetNew()/DecRef() pair into a "using" type function, so the DecRef() is guaranteed. Something like:
    let withNew (f: stream -> 'a) ms = 
        let x = ms.GetNew() 
        let ret = f x
        ms.DecRef()
        ret